nipoppy/workflow/proc_pipe/tractoflow/run_tractoflow.py

```diff
@@ -636,41 +661,64 @@ def run(participant_id, global_configs, session_id, output_dir, use_bids_filter,
 
     ## path to pipelines
     TRACTOFLOW_PIPE=f'{DATASET_ROOT}/workflow/proc_pipe/tractoflow'
-    
+
     ## nextflow arguments for logging - this is fixed for every run
     NEXTFLOW_CMD=f"nextflow -log {LOGDIR}/{participant_id}_ses-{session_id}_nf-log.txt run /scilus_flows/tractoflow/main.nf -work-dir {tractoflow_work_dir} -with-trace {LOGDIR}/{participant_id}_ses-{session_id}_nf-trace.txt -with-report {LOGDIR}/{participant_id}_ses-{session_id}_nf-report.html"
-    
+
     ## compose tractoflow arguments
     TRACTOFLOW_CMD=f""" --input {tractoflow_nxtf_inp} --output_dir {tractoflow_out_dir} --run_t1_denoising --run_gibbs_correction --encoding_direction {phase} --readout {readout} --dti_shells "0 {dti_use}" --fodf_shells "0 {odf_use}" --sh_order {sh_order} --profile {profile} --processes {ncore}"""
 
     ## TractoFlow arguments can be printed multiple ways that appear consistent with the documentation but are parsed incorrectly by nextflow.
     ## .nexflow.log (a run log that documents what is getting parsed by nexflow) shows additional quotes being added around the dti / fodf parameters. Something like: "'0' '1000'"
     ## Obviously, this breaks the calls that nextflow tries to make at somepoint (typically around Normalize_DWI) because half the command becomes an unfinished text block from the mysteriously added quotes.
     ## I don't know if the problem is python printing unhelpful/inaccurate text to the user or if nextflow can't parse its own input arguments correctly.
-    
+
     ## add resume option if working directory is not empty
     if not len(os.listdir(tractoflow_work_dir)) == 0:
         TRACTOFLOW_CMD = TRACTOFLOW_CMD + " -resume"
-    
+
     ## build command line call
-    CMD_ARGS = NEXTFLOW_CMD + TRACTOFLOW_CMD 
-    
+    CMD_ARGS = NEXTFLOW_CMD + TRACTOFLOW_CMD
+
     ## log what is called
     logger.info("-"*75)
     logger.info(f"Running TractoFlow for participant: {participant_id}")
 
-    ## singularity 
+    ## singularity
     SINGULARITY_CMD=f"{SINGULARITY_COMMAND} exec --cleanenv -H {nextflow_logdir} -B {nextflow_logdir}:/nextflow -B {LOGDIR} -B {output_dir} {SINGULARITY_TRACTOFLOW}"
 
     CMD=SINGULARITY_CMD + " " + CMD_ARGS
     logger.info("+"*75)
     logger.info(f"Command passed to system:\n\n{CMD}\n")
     logger.info("+"*75)
-    
+
     ## there's probably a better way to try / catch the .run() call here
     try:
         logger.info('Attempting Run')
         tractoflow_proc = subprocess.run(CMD, shell=True)
+
+        # do clean-up
```

Comment:
I would probably move this entire thing outside the `try`/`except` block

Reply:
Moved out.